monitoring/kernel.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoring
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gravitational/satellite/agent/health"
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	kernelCheckerID = "kernel-check"
+)
+
+// kernelChecker is a checker that verifies that the linux kernel version is
+// supported by Gravity.
+type kernelChecker struct {
+	// MinKernelVersion specifies the minimum supported kernel version.
+	MinKernelVersion KernelVersion
+}
+
+// NewKernelChecker returns a new instance of kernel checker.
+func NewKernelChecker(version KernelVersion) *kernelChecker {
+	return &kernelChecker{MinKernelVersion: version}
+}
+
+// Name returns the checker name.
+// Implements health.Checker.
+func (r *kernelChecker) Name() string {
+	return kernelCheckerID
+}
+
+// Check verifies kernel version.
+// Implements health.Checker.
+func (r *kernelChecker) Check(ctx context.Context, reporter health.Reporter) {
+	if err := r.check(ctx, reporter); err != nil {
+		log.WithError(err).Debug("Failed to verify kernel version.")
+		return
+	}
+	if reporter.NumProbes() == 0 {
+		reporter.Add(r.successProbe())
+	}
+}
+
+func (r *kernelChecker) check(_ context.Context, reporter health.Reporter) error {
+	release, err := realKernelVersionReader()
+	if err != nil {
+		return trace.Wrap(err, "failed to read kernel version")
+	}
+
+	kernelVersion, err := parseKernelVersion(release)
+	if err != nil {
+		reporter.Add(r.warningProbe(fmt.Sprintf("Failed to determine kernel version: %v.", release), err))
```

Comment:
Should this be a warning? I think we should only log this as an error and not create a probe.